### PR TITLE
Adding S3 object store support to Horovod template

### PR DESCRIPTION
**Pull request by CraigSterrett**
I have added support for S3 object store to the Horovod template.
Tested against both Minio and AWS S3, and validated the initial
localhost drive mappings still function properly.

fixes #{issue_num_here_if_applicable}

If you aren't done, please add the `wip` label until you're ready for review.

If you are adding a new template or mlt command/flag ensure that is is documented in the [docs/features.md](https://github.com/IntelAI/mlt/tree/master/docs/features.md) file.

## Reviewer Checklist

 - [ ] Do you understand it?
 - [ ] Is it correct?
 - [ ] Is it safe?
 - [ ] Is it legally compliant?
 - [ ] Is it robust?
 - [ ] Is it simple?
 - [ ] Is it tested?
 - [ ] Is it documented?
 - [ ] Is the style consistent?

See [the reviewer guide](docs/reviews.md) for more detail on each check.

### CODE REVIEW

**Comment by ashahba** [comment]
@CraigSterrett thanks for the PR.
We'll go over this and getting tests to pass early next week.

File: mlt-templates/horovod/README.md [hunk 1]
```diff
@@ -13,14 +13,37 @@ Template structure as follows:
...
+  * Can be updated with the command 
+
+    mlt template_config set template_parameters.output_path << hostpath >>
+
+## S3 object storage support
+
+We also have support for S3 object storage in the template.
+
+You will need to set S3 configuration parameters in the mlt.json file using the following commands
+
```
[the remark this answers is not in the document]
Comment (CraigSterrett) [reply mlt-templates/horovod/README.md]: Will include an updated parmeters.json in my update.  Will keep the readme as we discussed

File: mlt-templates/horovod/deploy.sh [hunk 2]
```diff
@@ -84,15 +84,20 @@ DATA_PATH=${DATA_PATH} # data path for training data
...
+AWS_SECRET_ACCESS_KEY=${AWS_SECRET_ACCESS_KEY} 
+AWS_REGION=${AWS_REGION} 
 
 EXEC="mpirun -np ${WORKERS} \
 --hostfile /kubeflow/openmpi/assets/hostfile \
 --map-by socket \
 -cpus-per-proc ${PHYSICAL_CORES} \
 --report-bindings \
---oversubscribe bash /src/app/exec_multiworker.sh ${PPR} ${NUM_INTER_THREADS} ${TOTAL_STEPS} ${LOG_STEPS} ${BATCH_SIZE} ${DATA_PATH} ${OUTPUT_PATH} ${NO_HOROVOD} ${LEARNING_RATE}"
+--oversubscribe bash /src/app/exec_multiworker.sh ${PPR} ${NUM_INTER_THREADS} ${TOTAL_STEPS} ${LOG_STEPS} ${BATCH_SIZE} ${DATA_PATH} ${OUTPUT_PATH} ${NO_HOROVOD} ${LEARNING_RATE} ${S3_ENDPOINT} ${S3_VERIFY_SSL} ${S3_USE_HTTPS} ${AWS_ACCESS_KEY_ID} ${AWS_SECRET_ACCESS_KEY} ${AWS_REGION}"
```
[the remark this answers is not in the document]
Comment (CraigSterrett) [reply mlt-templates/horovod/deploy.sh]: Agree, will include in my update

File: mlt-templates/horovod/deploy.sh [hunk 3]
```diff
@@ -84,15 +84,20 @@ DATA_PATH=${DATA_PATH} # data path for training data
 OUTPUT_PATH=${OUTPUT_PATH} # output path to store results
 NO_HOROVOD=${NO_HOROVOD}
 LEARNING_RATE=${LEARNING_RATE}
-
 PPR=$(( $NUM_WORKERS_PER_NODE / $SOCKETS_PER_NODE ))
+S3_ENDPOINT=${S3_ENDPOINT}
```
[the remark this answers is not in the document]
Comment (CraigSterrett) [reply mlt-templates/horovod/deploy.sh]: Agree will include in my update

**Comment by ashahba** [comment]
@CraigSterrett I have added you as contributor to the repo.
Please feel free to return the CI tests

**Pull request closed without merging**